linkerd/app/inbound/src/policy.rs

```diff
@@ -16,23 +16,21 @@ use linkerd_app_core::{
     transport::{ClientAddr, OrigDstAddr, Remote},
     Result,
 };
+use linkerd_cache::Cached;
 pub use linkerd_server_policy::{Authentication, Authorization, Protocol, ServerPolicy, Suffix};
 use thiserror::Error;
 use tokio::sync::watch;
 
 #[derive(Clone, Debug, Error)]
-#[error("unauthorized connection on unknown port {0}")]
-pub struct DeniedUnknownPort(pub u16);
-
-#[derive(Clone, Debug, Error)]
-#[error("unauthorized connection on server {server}")]
+#[error("unauthorized connection on {kind}/{name}")]
 pub struct DeniedUnauthorized {
-    server: std::sync::Arc<str>,
+    kind: std::sync::Arc<str>,
+    name: std::sync::Arc<str>,
 }
 
-pub trait CheckPolicy {
+pub trait GetPolicy {
     /// Checks that the destination address is configured to allow traffic.
```

Comment:
doc comment should maybe be changed to 
```suggestion
    /// Returns the allowed traffic policy configured for the destination address.
```
or similar?